## Sigmoid function

In [1]:
from math import e

In [3]:
def sigmoid(z):
    return (1/(1+e**-z))

In [4]:
sigmoid(.0001)

0.5000249999999792

In [5]:
sigmoid(100000)

1.0

In [7]:
sigmoid(-100)

3.7200759760208555e-44

## Softmax function

* Suppose we have three classes in the output 
* After calculating the z value for each output we get these values (z=w.x+b)

#### First Step: Calculate the exp of each output

In [21]:
from math import exp

z = [-1, 1, 5]

for i in z:
    print(exp(i))


0.36787944117144233
2.718281828459045
148.4131591025766


#### Second Step: Sum up the exponentials

In [22]:
total = sum(exp(i) for i in z)
total

151.49932037220708

#### Third Step: Calculate the proportion for each class

In [24]:
for i in z:
    print ((exp(i)/total)*100)

0.24282580295913375
1.7942534803329195
97.96292071670794


**Interpretation:**
* The input has 0.2% probability to match the first class
* The input has 1.8% probability to match the second class
* The input has 97.96% probability to match the third class